src/lib/parser/core/PokemonSaveParser.ts

```diff
@@ -613,6 +618,68 @@ export class PokemonSaveParser {
     this.webSocketClient.addMemoryChangeListener(memoryChangeListener)
   }
 
+  /**
+   * Create SaveData from memory update instead of refetching all data
+   * Uses the passed memory data to construct party information efficiently
+   */
+  private async createSaveDataFromMemoryUpdate (address: number, _size: number, data: Uint8Array): Promise<SaveData> {
```

Comment:
refactor this function to be more simple an concise, and patch the data we already have using it, then send the patched data to the parser. if you need to rewrite, refactor some of the parser structure to achieve that, do that. bot avoid calling any readMemory here, we only want to patch data here, no additional reading